src/oss/contributing/documentation.mdx

```diff
@@ -0,0 +1,358 @@
+
+<Steps>
+    <Step title="Find the page">
+        {/* TODO */}
+        Navigate to any documentation page and click the "edit this page" link at the bottom
+    </Step>
+
+    <Step title="Fork the repository">
+        GitHub will prompt you to fork the repository to your account. Make sure to fork into your personal account, **not an organization**
+    </Step>
+
+    <Step title="Make your changes">
+        Correct the typo directly in GitHub's editor
+    </Step>
+
+    <Step title="Commit your changes">
+        Click `Commit changes...` and give your commit a descriptive title like `fix(docs): description`
+    </Step>
+
+    <Step title="Create pull request">
+        GitHub will redirect you to create a pull request. Title it like `docs: Fix typo in X` and follow the PR template checklist
+    </Step>
+</Steps>
+
+<Note>
+    Docs PRs are typically reviewed within a few days. Keep an eye on your PR to address any feedback from maintainers.
+</Note>
+
+### Full development setup
+
+For larger changes or ongoing contributions, set up a local development environment:
+
+<Steps>
+    <Step title="Local repo">
+        :::python
+        Clone our [documentation repository](https://github.com/langchain-ai/docs) unless you plan to update API references only, in which case you can fork the relevant repo (e.g. [`langchain`](https://github.com/langchain-ai/langchain) or [`langgraph`](https://github.com/langchain-ai/langgraph)).
+        :::
+        :::js
+        Clone our [documentation repository](https://github.com/langchain-ai/docs) unless you plan to update API references only, in which case you can fork the relevant repo (e.g. [`langchainjs`](https://github.com/langchain-ai/langchainjs) or [`langgraphjs`](https://github.com/langchain-ai/langgraphjs)).
+        :::
+    </Step>
+
+    <Step title="Development environment">
+        Set up your development environment following our [IDE setup guide](#ide-setup)
+    </Step>
+
+    <Step title="Understanding the build system">
+        Familiarize yourself with our documentation [build pipeline](#build-system)
+    </Step>
+
+    <Step title="Mintlify knowledge">
+        Review the [Mintlify documentation](https://mintlify.com/docs) for component usage and best practices
+    </Step>
+</Steps>
+
+### IDE setup
+
+This repository includes configuration files to ensure consistent formatting across different editors:
+
+<Tabs>
+    <Tab title="VSCode">
+        Settings in `.vscode/settings.json` are automatically applied when you open the project. No additional setup required.
+
+        **Recommended extensions:**
+        - [EditorConfig for VS Code](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig)
+        - [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one)
+        - [MDX](https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx)
```

Comment:
Should we have an `extensions.json` file to allow developers to auto-install these recommendations?